### PR TITLE
THORN-2182: MP Rest Client connection leak test

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
- [X] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [X] [v2] Have you created a [JIRA](https://issues.jboss.org/browse/THORN) and used it in the commit message?
- [ ] [v4] Have you created a [GitHub Issue](https://github.com/thorntail/thorntail/issues) and used it in the commit message?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/thorntail/thorntail/pulls) for the same issue?
- [ ] Have you built the project locally prior to submission with `mvn clean install`?

-----
Added a test to verify if MP Rest Client connections don't leak on failure

### CODE REVIEW

**Comment by michalszynkiewicz** [comment]
**NOTE** requires https://github.com/smallrye/smallrye-rest-client/pull/12

**Comment by michalszynkiewicz** [comment]
retest this please

File: testsuite/testsuite-microprofile-restclient/src/main/java/org/wildfly/swarm/microprofile/restclient/FailOnFailOnEvenRequestsResource.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+/**
+ * @author Michal Szynkiewicz, michal.l.szynkiewicz@gmail.com
+ * <br>
+ * Date: 9/27/18
+ */
+@Path("/")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class FailOnFailOnEvenRequestsResource {
```
Comment (Ladicek) [inline testsuite/testsuite-microprofile-restclient/src/main/java/org/wildfly/swarm/microprofile/restclient/FailOnFailOnEvenRequestsResource.java]: Double `FailOn`?

**Comment by Ladicek** [comment]
Who will do the SmallRye RestClient release? :-)

File: testsuite/testsuite-microprofile-restclient/src/main/java/org/wildfly/swarm/microprofile/restclient/FailOnFailOnEvenRequestsResource.java (hunk 1 above)
Comment (michalszynkiewicz) [reply testsuite/testsuite-microprofile-restclient/src/main/java/org/wildfly/swarm/microprofile/restclient/FailOnFailOnEvenRequestsResource.java]: fixed

**Comment by Ladicek** [comment]
The Linux test failure seems to be caused by SmallRye RestClient 1.0.1 not yet being available in Maven Central.

**Comment by michalszynkiewicz** [comment]
retest this please

**Comment by Ladicek** [comment]
Windows failure (MP FT TCK) is unrelated.

**Review by Ladicek — APPROVED** [review]

**Pull request merged**